### DISCUSSION TITLE
Iphone 3gs white screen

### DISCUSSION THREAD

**Discussion by kman82** [LukeZGD/Legacy-iOS-Kit#338] [Q&A] [answered]
Hi all,

Ive had an issue after jailbreaking and loading cfw via redsnow. 

All of a sudden the display is white screen. I have been able to get in via ssh using the Legacy IOS kit as I would like to wipe the phone but as soon as I try to mount the hard drive it just locks up.

Waiting for disks.....

Does this mean the disk is no good now?

trying to restore also outputs:

unable to find AppleNANDFTL

Anyone got any other options I can try?

** Legacy iOS Kit ***
 - Script by LukeZGD -

* Version: v23.10.26 (b14db3a)
* Platform: linux (Ubuntu 23.04) 

* Device: iPhone2,1 (n88ap) in DFU mode
* This iPhone2,1 is a new bootrom model
* Cannot check for manufacturing date in DFU mode
* iOS Version: Unknown
* To get iOS version, go to: Other Utilities -> Get iOS Version
* ECID: 3768734518573

* Selected IPSW: ../iPhone2,1_4.1_8B117_Restore.ipsw
* Hacktivation is supported for this restore

 > Main Menu > Restore/Downgrade > 4.1
[Input] Select an option:
1) Select Target IPSW	 3) Start Restore
2) Download Target IPSW	 4) Go Back
#? 3
[Input] Jailbreak Option
* When this option is enabled, your device will be jailbroken on restore.
* I recommend to enable this option to have the jailbreak and Cydia pre-installed.
* This option is enabled by default (Y). Select this option if unsure.
[Input] Enable this option? (Y/n): y
[Log] Jailbreak option enabled.

[Input] Hacktivate Option
* When this option is enabled, your device will be activated on restore.
* Enable this option if you have no valid SIM card to activate the phone.
* This option is disabled by default (N). Select this option if unsure.
[Input] Enable this option? (y/N): y
[Log] Hacktivate option enabled by user.

[Log] Found existing Custom IPSW. Skipping IPSW creation.
[Log] Found existing saved 4.1 blobs: ../saved/shsh/3768734518573_iPhone2,1_n88ap_4.1-8B117_3a88b7c3802f2f0510abc432104a15ebd8bd7154.shsh2
[Log] Placing device to pwnDFU mode using ipwnder
device is already in pwned DFU mode. not executing exploit.
[Log] Device iPhone2,1 has no baseband/disabled baseband update
[Log] Extracting IPSW: ../iPhone2,1_4.1_8B117_CustomHJ.ipsw
Archive:  ../iPhone2,1_4.1_8B117_CustomHJ.ipsw
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/018-7061-122.dmg  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/018-7073-079.dmg  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/018-7080-079.dmg  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/BuildManifest.plist  
   creating: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/
   creating: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/dfu/
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/dfu/iBEC.n88ap.RELEASE.dfu  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/dfu/iBSS.n88ap.RELEASE.dfu  
   creating: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/
   creating: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/applelogo.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/recoverymode.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/glyphplugin.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/glyphcharging.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/batteryfull.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/LLB.n88ap.RELEASE.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/batterycharging1.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/iBoot.n88ap.RELEASE.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/batterylow1.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/batterylow0.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/manifest  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/DeviceTree.n88ap.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Firmware/all_flash/all_flash.n88ap.production/batterycharging0.s5l8920x.img3  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/kernelcache.release.n88  
 extracting: ../iPhone2,1_4.1_8B117_CustomHJ/Restore.plist  
[Log] Running idevicerestore with command: sudo ../bin/linux/x86_64/idevicerestore -ew "../iPhone2,1_4.1_8B117_CustomHJ.ipsw"
Found device in DFU mode
Identified device as n88ap, iPhone2,1
Extracting BuildManifest from IPSW
Product Version: 4.1
Product Build: 8B117 Major: 8
Device supports Image4: false
Variant: Customer Erase Install (IPSW)
This restore will erase your device data.
Found ECID 3768734518573
checking for local shsh
Using cached SHSH
Using cached filesystem from '../iPhone2,1_4.1_8B117_CustomHJ/018-7061-122.dmg'
Extracting iBSS.n88ap.RELEASE.dfu...
Personalizing IMG3 component iBSS...
reconstructed size: 108785
Sending iBSS (108785 bytes)...
[==================================================] 100.0%
INFO: device serial number is 86947ZUE3NP
Recovery Mode Environment:
iBoot build-version=iBoot-931.18.27
iBoot build-style=RELEASE
ramdisk-size=RELEASE
Extracting 018-7080-079.dmg...
Personalizing IMG3 component RestoreRamDisk...
reconstructed size: 16984233
Sending RestoreRamDisk (16984233 bytes)...
Extracting DeviceTree.n88ap.img3...
Personalizing IMG3 component RestoreDeviceTree...
reconstructed size: 48153
Sending RestoreDeviceTree (48153 bytes)...
Extracting kernelcache.release.n88...
Personalizing IMG3 component RestoreKernelCache...
reconstructed size: 4761433
Sending RestoreKernelCache (4761433 bytes)...
About to restore device... 
Waiting for device...
Device 376ba7be823f8962602d786e29182389903d650a is now connected in restore mode...
Connecting now...
Connected to com.apple.mobile.restored, version 12
Device 376ba7be823f8962602d786e29182389903d650a has successfully entered restore mode
Hardware Information:
BoardID: 0
ChipID: 35104
UniqueChipID: 3768734518573
ProductionMode: true
Starting FDR listener thread
ERROR: Unable to connect to FDR client (-2)
ERROR: Failed to start FDR Ctrl channel
Got status message
Unhandled status message (28)
common.c:printing 6277 bytes plist:
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>AMRError</key>
	<integer>28</integer>
	<key>Log</key>
	<string>n0CIS(): ProductInfo0: "s=D1"
IOSDIOIoCardDevice::parseFn0CIS(): ProductInfo1: "P=N88 m=3.2"
AppleBCMWLAN::init(): AppleBCMWLAN-44.10.3 Aug  4 2010 22:51:47
AppleBCMWLAN::init(): Starting with debug level: 4, debug flags: 00000000
AppleBCMWLAN::init(): AppleBCMWLAN-44.10.3 Aug  4 2010 22:51:47
AppleBCMWLAN::init(): Starting with debug level: 4, debug flags: 00000000
found suitable IOMobileFramebuffer: AppleM2CLCD
display-scale = 1
display-rotation = 0
found PTP interface
AppleSynopsysOTGDevice - Configuration: PTP
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice - Configuration: iPod USB Interface
AppleSynopsysOTGDevice          Interface: USBAudioControl
AppleSynopsysOTGDevice          Interface: USBAudioStreaming
AppleSynopsysOTGDevice          Interface: IapOverUsbHid
AppleSynopsysOTGDevice - Configuration: PTP + Apple Mobile Device
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice          Interface: AppleUSBMux
AppleSynopsysOTGDevice - Configuration: PTP + Apple Mobile Device + Apple USB Ethernet
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice          Interface: AppleUSBMux
AppleSynopsysOTGDevice          Interface: AppleUSBEthernet
AppleSynopsysOTGDevice::gated_registerFunction Register function USBAudioControl
AppleSynopsysOTGDevice::gated_registerFunction Register function USBAudioStreaming
IOAccessoryPortUSB::start
AppleSynopsysOTGDevice::gated_registerFunction Register function IapOverUsbHid
virtual bool AppleUSBDeviceMux::start(IOService*) build: Aug  4 2010 22:48:34
init_waste
AppleSynopsysOTGDevice::gated_registerFunction Register function AppleUSBMux
AppleSynopsysOTGDevice::gated_registerFunction Register function AppleUSBEthernet
AppleUSBEthernetDevice::start: Host MAC address = 64:b9:e8:8b:bd:4d
AppleSynopsysOTGDevice::gated_registerFunction Register function PTP
AppleSynopsysOTGDevice::gated_registerFunction all functions registered- we are ready to start usb stack
AppleS5L8720XUSBPhy::powerUp : _powered: 0
AppleEmbeddedUSBArbitrator::enableDeviceClock : enable: 1, index: 0
IOReturn AppleUSBDeviceMux::setPropertiesGated(OSObject*) setting debug level to 7
AppleSynopsysOTGDevice::handleUSBReset
AppleSynopsysOTGDevice::handleUSBReset
unrecognized request 'GetValue'
unrecognized request 'GetValue'
close(caller = 0x51df): remote port = 49152
unrecognized request 'GetValue'
AppleUSBDeviceMux::handleConnectResult new session 0x83714280 established 62078&lt;-lo0-&gt;49152 62078&lt;-usb-&gt;1
AppleUSBDeviceMux::handleConnectResult new session 0x83714380 established 62078&lt;-lo0-&gt;49153 62078&lt;-usb-&gt;2
AppleUSBDeviceMux::handleConnectResult new session 0x83714700 established 62078&lt;-lo0-&gt;49154 62078&lt;-usb-&gt;3
void AppleUSBDeviceMux::handleMuxTCPInput(__mbuf*) received reset, closing 0x83714280
AppleUSBEthernetDevice::message: kIOUSBDeviceInterfaceMessageTypeSetAlternateSetting, new alt-setting 1
AppleUSBDeviceMux::handleConnectResult new session 0x825cfb00 established 62078&lt;-lo0-&gt;49155 62078&lt;-usb-&gt;4
AppleUSBDeviceMux::handleConnectResult new session 0x83714800 established 62078&lt;-lo0-&gt;49156 62078&lt;-usb-&gt;5
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
recv(6, 4) failed: connection closed
unable to read message size: -1
could not receive message
close(caller = 0x51df): remote port = 49155
void AppleUSBDeviceMux::handleMuxTCPInput(__mbuf*) received reset, closing 0x825cfb00
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
client protocol version 12
Restore options:
	UpdateBaseband                 =&gt; &lt;CFBoolean 0x1ac7bc [0x1abd60]&gt;{value = false}
	SystemPartitionSize            =&gt; &lt;CFNumber 0xb08230 [0x1abd60]&gt;{value = +800, type = kCFNumberSInt64Type}
entering partition_nand_device
device is not a boot-from-NAND device
entering wait_for_storage_device
waiting for matching IOKit service: &lt;CFBasicHash 0xb084a0 [0x1abd60]&gt;{type = mutable dict, count = 1,
entries =&gt;
	2 : &lt;CFString 0x2c3df0 [0x1abd60]&gt;{contents = "IOProviderClass"} = &lt;CFString 0xb08740 [0x1abd60]&gt;{contents = "AppleNANDFTL"}
}

void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
waiting for matching IOKit service: &lt;CFBasicHash 0xb084a0 [0x1abd60]&gt;{type = mutable dict, count = 1,
entries =&gt;
	2 : &lt;CFString 0x2c3df0 [0x1abd60]&gt;{contents = "IOProviderClass"} = &lt;CFString 0xb08740 [0x1abd60]&gt;{contents = "AppleNANDFTL"}
}

waiting for matching IOKit service: &lt;CFBasicHash 0xb084a0 [0x1abd60]&gt;{type = mutable dict, count = 1,
entries =&gt;
	2 : &lt;CFString 0x2c3df0 [0x1abd60]&gt;{contents = "IOProviderClass"} = &lt;CFString 0xb08740 [0x1abd60]&gt;{contents = "AppleNANDFTL"}
}

waiting for matching IOKit service: &lt;CFBasicHash 0xb084a0 [0x1abd60]&gt;{type = mutable dict, count = 1,
entries =&gt;
	2 : &lt;CFString 0x2c3df0 [0x1abd60]&gt;{contents = "IOProviderClass"} = &lt;CFString 0xb08740 [0x1abd60]&gt;{contents = "AppleNANDFTL"}
}

unable to find AppleNANDFTL
</string>
	<key>MsgType</key>
	<string>StatusMsg</string>
	<key>Status</key>
	<integer>28</integer>
</dict>
</plist>
Log is available:
n0CIS(): ProductInfo0: "s=D1"
IOSDIOIoCardDevice::parseFn0CIS(): ProductInfo1: "P=N88 m=3.2"
AppleBCMWLAN::init(): AppleBCMWLAN-44.10.3 Aug  4 2010 22:51:47
AppleBCMWLAN::init(): Starting with debug level: 4, debug flags: 00000000
AppleBCMWLAN::init(): AppleBCMWLAN-44.10.3 Aug  4 2010 22:51:47
AppleBCMWLAN::init(): Starting with debug level: 4, debug flags: 00000000
found suitable IOMobileFramebuffer: AppleM2CLCD
display-scale = 1
display-rotation = 0
found PTP interface
AppleSynopsysOTGDevice - Configuration: PTP
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice - Configuration: iPod USB Interface
AppleSynopsysOTGDevice          Interface: USBAudioControl
AppleSynopsysOTGDevice          Interface: USBAudioStreaming
AppleSynopsysOTGDevice          Interface: IapOverUsbHid
AppleSynopsysOTGDevice - Configuration: PTP + Apple Mobile Device
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice          Interface: AppleUSBMux
AppleSynopsysOTGDevice - Configuration: PTP + Apple Mobile Device + Apple USB Ethernet
AppleSynopsysOTGDevice          Interface: PTP
AppleSynopsysOTGDevice          Interface: AppleUSBMux
AppleSynopsysOTGDevice          Interface: AppleUSBEthernet
AppleSynopsysOTGDevice::gated_registerFunction Register function USBAudioControl
AppleSynopsysOTGDevice::gated_registerFunction Register function USBAudioStreaming
IOAccessoryPortUSB::start
AppleSynopsysOTGDevice::gated_registerFunction Register function IapOverUsbHid
virtual bool AppleUSBDeviceMux::start(IOService*) build: Aug  4 2010 22:48:34
init_waste
AppleSynopsysOTGDevice::gated_registerFunction Register function AppleUSBMux
AppleSynopsysOTGDevice::gated_registerFunction Register function AppleUSBEthernet
AppleUSBEthernetDevice::start: Host MAC address = 64:b9:e8:8b:bd:4d
AppleSynopsysOTGDevice::gated_registerFunction Register function PTP
AppleSynopsysOTGDevice::gated_registerFunction all functions registered- we are ready to start usb stack
AppleS5L8720XUSBPhy::powerUp : _powered: 0
AppleEmbeddedUSBArbitrator::enableDeviceClock : enable: 1, index: 0
IOReturn AppleUSBDeviceMux::setPropertiesGated(OSObject*) setting debug level to 7
AppleSynopsysOTGDevice::handleUSBReset
AppleSynopsysOTGDevice::handleUSBReset
unrecognized request 'GetValue'
unrecognized request 'GetValue'
close(caller = 0x51df): remote port = 49152
unrecognized request 'GetValue'
AppleUSBDeviceMux::handleConnectResult new session 0x83714280 established 62078<-lo0->49152 62078<-usb->1
AppleUSBDeviceMux::handleConnectResult new session 0x83714380 established 62078<-lo0->49153 62078<-usb->2
AppleUSBDeviceMux::handleConnectResult new session 0x83714700 established 62078<-lo0->49154 62078<-usb->3
void AppleUSBDeviceMux::handleMuxTCPInput(__mbuf*) received reset, closing 0x83714280
AppleUSBEthernetDevice::message: kIOUSBDeviceInterfaceMessageTypeSetAlternateSetting, new alt-setting 1
AppleUSBDeviceMux::handleConnectResult new session 0x825cfb00 established 62078<-lo0->49155 62078<-usb->4
AppleUSBDeviceMux::handleConnectResult new session 0x83714800 established 62078<-lo0->49156 62078<-usb->5
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
recv(6, 4) failed: connection closed
unable to read message size: -1
could not receive message
close(caller = 0x51df): remote port = 49155
void AppleUSBDeviceMux::handleMuxTCPInput(__mbuf*) received reset, closing 0x825cfb00
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
client protocol version 12
Restore options:
	UpdateBaseband                 => <CFBoolean 0x1ac7bc [0x1abd60]>{value = false}
	SystemPartitionSize            => <CFNumber 0xb08230 [0x1abd60]>{value = +800, type = kCFNumberSInt64Type}
entering partition_nand_device
device is not a boot-from-NAND device
entering wait_for_storage_device
waiting for matching IOKit service: <CFBasicHash 0xb084a0 [0x1abd60]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x2c3df0 [0x1abd60]>{contents = "IOProviderClass"} = <CFString 0xb08740 [0x1abd60]>{contents = "AppleNANDFTL"}
}

void AppleUSBDeviceMux::handleConnectResult(BulkUSBMuxSession*, errno_t) new session to port 1082 failed: 61
waiting for matching IOKit service: <CFBasicHash 0xb084a0 [0x1abd60]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x2c3df0 [0x1abd60]>{contents = "IOProviderClass"} = <CFString 0xb08740 [0x1abd60]>{contents = "AppleNANDFTL"}
}

waiting for matching IOKit service: <CFBasicHash 0xb084a0 [0x1abd60]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x2c3df0 [0x1abd60]>{contents = "IOProviderClass"} = <CFString 0xb08740 [0x1abd60]>{contents = "AppleNANDFTL"}
}

waiting for matching IOKit service: <CFBasicHash 0xb084a0 [0x1abd60]>{type = mutable dict, count = 1,
entries =>
	2 : <CFString 0x2c3df0 [0x1abd60]>{contents = "IOProviderClass"} = <CFString 0xb08740 [0x1abd60]>{contents = "AppleNANDFTL"}
}

unable to find AppleNANDFTL

ERROR: Unable to successfully restore device
ERROR: Unable to restore device

[Log] Restoring done! Read the message below if any error has occurred:
* For device activation, go to: Other Utilities -> Attempt Activation
* If you are getting the error "could not retrieve device serial number":
 -> This means that your device is not compatible with 4.1
* If the restore failed on updating baseband:
 -> Try disabling baseband update: ./restore.sh --disable-bbupdate

* Please read the "Troubleshooting" wiki page in GitHub before opening any issue!
* Your problem may have already been addressed within the wiki page.
* If opening an issue in GitHub, please provide a FULL log/output. Otherwise, your issue may be dismissed.
[WARNING] Do not uninstall Cydia Substrate and Substrate Safe Mode in Cydia!

**Comment by LukeZGD** [comment, marked as answer]
Very late reply, but this means that your device is likely not compatible with 4.1
